addons/sale_purchase/models/sale_order.py

```diff
@@ -0,0 +1,323 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from dateutil.relativedelta import relativedelta
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+from odoo.tools import float_compare
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    purchase_order_count = fields.Integer("Number of Purchase Order", compute='_compute_purchase_order_count')
+
+    @api.multi
+    def _compute_purchase_order_count(self):
+        self.env.cr.execute("""
+            SELECT SOL.order_id, COUNT(DISTINCT POL.order_id) as purchase_count
+            FROM sale_order_line SOL
+                JOIN purchase_order_line POL ON POL.sale_line_id = SOL.id
+            WHERE SOL.order_id IN %s
+            GROUP BY SOL.order_id;
```

Comment:
Are you sure you won't have inconsistencies in multi-company ? It will count all the purchase orders, even those you can't access via the ir.rules.